lib/rules/valid-local-dependency.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Checks existence of local dependencies in the package.json
+ * @author Kendall Gassner
+ */
+'use strict';
+const path = require('path');
+const {
+    isPackageJson,
+    extractPackageObjectFromAST
+} = require('../processors/PackageJsonProcessor');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                'Checks existence of local dependencies in the package.json',
+            category: 'Best Practices',
+            recommended: true
+        },
+        fixable: null // or "code" or "whitespace"
+    },
+
+    create: function(context) {
+        return {
+            'Program:exit': node => {
+                if (!isPackageJson(context.getFilename())) {
+                    return;
+                }
+                const sourceCode = context.getSourceCode();
+                const packageRoot = extractPackageObjectFromAST(node);
+                const original = JSON.parse(sourceCode.getText(packageRoot));
+                const {
+                    dependencies,
+                    peerDependencies,
+                    devDependencies
+                } = original;
+
+                const depObjs = [
+                    Object.entries(dependencies || {}),
+                    Object.entries(peerDependencies || {}),
+                    Object.entries(devDependencies || {})
+                ];
+
+                depObjs.map(obj =>
+                    obj.map(([key, value]) => {
```

Comment:
[Nitpicking] These would be a little more semantically correct as `forEach` (https://eslint.org/docs/latest/rules/array-callback-return). Definitely not a blocker IMO 😄 

```suggestion
                depObjs.forEach(obj =>
                    obj.forEach(([key, value]) => {
```